agent/go.mod

```diff
@@ -29,6 +29,223 @@ require (
 	gotest.tools v2.2.0+incompatible
 )
 
+require (
```

Comment:
For posterity, anyone wondering what's up with this, from the Go 1.17 release notes:

> For the go command to correctly resolve transitive imports using the pruned module graph, the go.mod file for each module needs to include more detail about the transitive dependencies relevant to that module. If a module specifies go 1.17 or higher in its go.mod file, its go.mod file now contains an explicit require directive for every module that provides a transitively-imported package. (In previous versions, the go.mod file typically only included explicit requirements for directly-imported packages.)

Reply:
I'd tack on https://golang.org/ref/mod is great.